third_party/move/move-compiler-v2/src/pipeline/flush_writes_processor.rs

```diff
@@ -250,78 +264,223 @@ impl FlushWritesProcessor {
                 // If used outside the basic block, flush right away.
                 return true;
             }
-            // If has intervening definitions, flush right away.
-            // The first call checks the definitions of preceding uses in the same instruction.
-            // The second call checks definitions between `def` and `use_`.
-            Self::has_intervening_def(&def, use_, use_def_links)
-                || Self::has_flush_causing_defs_in_between(
-                    &def,
-                    use_,
-                    code,
-                    use_def_links,
-                    flush_writes,
-                )
+            false
         })
     }
 
-    /// Given the `use_` of `def`, is there a previous use of any temp at the same
-    /// instruction as `use_`, which has a definition after `def` and before
-    /// the `use_` instruction?
-    fn has_intervening_def(
-        def: &DefOrUsePoint,
-        use_: &DefOrUsePoint,
+    /// Given an instruction at `offset` of `code`, check if there are uses of temporaries in the
+    /// wrong order, and if so, mark the corresponding temporaries for flushing.
```

Comment:
Maybe its helpful to quickly describe what "wrong order" means, or reference to some place in the module documentation.